clients/filesystem-fuse/src/filesystem.rs

```diff
@@ -0,0 +1,236 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+use async_trait::async_trait;
+use bytes::Bytes;
+use fuse3::{Errno, FileType, Timestamp};
+
+pub(crate) type Result<T> = std::result::Result<T, Errno>;
+
+/// RawFileSystem interface for the file system implementation. it use by FuseApiHandle,
+/// it ues the file id to operate the file system apis
+/// the `file_id` and `parent_file_id` it is the unique identifier for the file system,
+/// it is used to identify the file or directory
+/// the `handle_id` it is the file handle, it is used to identify the opened file,
+/// it is used to read or write the file content
+/// the `file id` and `handle_id` need to mapping the `ino`/`inode` and `fh` in the fuse3
+#[async_trait]
+pub(crate) trait RawFileSystem: Send + Sync {
+    /// Init the file system
+    async fn init(&self);
+
+    /// Get the file path by file id
+    async fn get_file_path(&self, file_id: u64) -> String;
+
+    /// Validate the file id and file handle, if the file id or file handle is invalid, return error
+    async fn valid_file_id(&self, file_id: u64, fh: u64) -> Result<()>;
+
+    /// Get the file stat by file id
+    async fn stat(&self, file_id: u64) -> Result<FileStat>;
+
+    /// Get the file stat by parent file id and file name
+    async fn lookup(&self, parent_file_id: u64, name: &str) -> Result<FileStat>;
+
+    /// Read the directory by file id
+    async fn read_dir(&self, dir_file_id: u64) -> Result<Vec<FileStat>>;
+
+    /// Open the file by file id and flags
+    async fn open_file(&self, file_id: u64, flags: u32) -> Result<FileHandle>;
```

Comment:
why return FileHandle not file id?

Reply:
open file should return the file handle id